.github/workflows/java-ci.yml

```diff
@@ -87,7 +87,7 @@ jobs:
       with:
         distribution: zulu
         java-version: 8
-    - run: ./gradlew -DallVersions build -x test -x javadoc -x integrationTest
+    - run: ./gradlew -DallVersions build -x test -x javadoc -x integrationTest --scan
```

Comment:
doesnt this require to agree to gradle's terms of service?
see https://docs.gradle.com/enterprise/gradle-plugin/#connecting_to_scans_gradle_com

imo if the sporadic errors always happen in checkstyle part, we may want to check if we can upgrade the respective plugin to fix the issue

Reply:
> doesnt this require to agree to gradle's terms of service
good point, in that case we probably cannot enable it

Reply:
I think other projects like Nessie Agree to these terms of service. I have skimmed it and didn't find anything harmful.

Do you guys think we should not enable `--scan` and just enable `--stacktrace` `--debug `?

Reply:
>  imo if the sporadic errors always happen in checkstyle part, we may want to check if we can upgrade the respective plugin to fix the issue

we are using the latest version I assume as I didn't find any version pinning. 
https://github.com/apache/iceberg/blob/master/baseline.gradle#L36

Reply:
we are only using latest version of baseline which supported java 8:
https://github.com/apache/iceberg/blob/35692a360ffd2d1b9107ec6018f9fd97fda04671/build.gradle#L30-L34

in there we can find the checkstyle version:
https://github.com/palantir/gradle-baseline/blob/4.42.0/versions.props#L17

but i think 9.3 would be the last version with java8 support:
https://checkstyle.org/releasenotes.html#Release_9.3

Reply:
I think it's more of a licensing issue, not sure if there is any other project under Apache Foundation using this feature, we can ping the infra team about this.

Reply:
True but the `gradle-baseline-4.42.0` may not accept new patches :( 

Also what is your opinion on 
> Do you guys think we should not enable --scan and just enable --stacktrace --debug ?

Reply:
we need to have the failure reason before raising the patch to the gradle-baseline. So, I believe we need to get stack trace/debug first.

Reply:
> before raising the patch to the gradle-baseline

i was trying to say there would probably be ways to bump the checkstyle version without needing to get a new gradle-baseline version

that being said, i agree we should try to get stacktrace first to see what exactly goes wrong occasionally